src/javatests/com/google/cloud/bigquery/dwhassessment/hooks/logger/DatePartitionedRecordsWriterFactoryTest.java

```diff
@@ -233,9 +183,30 @@ private static Instant parseDateTime(String dateTimeString) {
     return Instant.parse(dateTimeString);
   }
 
-  private String getCurrentWriterFolder(DatePartitionedRecordsWriterFactory recordsWriterFactory)
+  private static ImmutableList<String> getDirectoryFiles(FileSystem fs, Path parent)
       throws IOException {
-    return recordsWriterFactory.createWriter().getPath().getParent().getName();
+    RemoteIterator<LocatedFileStatus> files = fs.listFiles(parent, true);
```

Comment:
Optional: I would suggest to use Guava to convert `Iterator` to `Stream` and then collect to `ImmutableList`.

```
return Streams.stream(fs.listFiles(parent, true))
  .map(file -> file.getPath().getName())
  .collect(toImmutableList());
```
https://guava.dev/releases/31.1-jre/api/docs/com/google/common/collect/Streams.html#stream(java.util.Iterator)

Reply:
I tried it in the beginning, but `listFiles` returns a RemoteIterator, which is a dedicated interface and can't be passed to guava methods